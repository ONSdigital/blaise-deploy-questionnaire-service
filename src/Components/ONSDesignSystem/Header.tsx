import React, {ReactElement} from "react";

interface Props {
    title: string
}


function Header(props: Props): ReactElement {
    return (
        <>
            <header className="header header--internal">
                <div className="header__top" role="banner">
                    <div className="container">
                        <div
                            className="header__grid-top grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap ">
                            <div className="grid__col col-auto">
                                <a className="header__logo-link" href="/">
                                    <picture>
                                        {/*<source media="(max-width: 499px)" srcSet="/img/ons-logo-stacked-neg-en.svg"*/}
                                        {/*        alt="Office for National Statistics logo"/>*/}
                                            <img className="header__logo" src="https://ons-design-system.netlify.app/img/ons-logo-neg-en.svg"
                                                 alt="Office for National Statistics logo"/>
                                    </picture>
                                </a>
                            </div>
                        </div>
                    </div>
                </div>
                <div className="header__main">
                    <div className="container">
                        <div
                            className="grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap">
                            <div className="grid__col col-auto u-flex-shrink">
                                <div className="header__title">{props.title}</div>
                            </div>
                        </div>
                    </div>
                </div>
            </header>
        </>
);
}

export default Header;
