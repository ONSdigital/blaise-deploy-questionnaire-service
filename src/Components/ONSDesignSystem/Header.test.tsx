import React from "react";
import Enzyme, { render, shallow} from "enzyme";

import Adapter from "enzyme-adapter-react-16";
import Header from "./Header";

describe("Header Test", () => {
    Enzyme.configure({adapter: new Adapter()});

    const Props = {
        title: "App Title"
    };

    it("matches Snapshot", () => {
        const wrapper = render(<Header title={Props.title}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render correctly", () => {
        const wrapper = shallow(<Header title={Props.title}/>);
        expect(wrapper.exists()).toEqual(true);
    });

    it("should render with the title displayed", () => {
        const wrapper = shallow(<Header title={Props.title}/>);
        expect(wrapper.html()).toContain(Props.title);
    });
});
