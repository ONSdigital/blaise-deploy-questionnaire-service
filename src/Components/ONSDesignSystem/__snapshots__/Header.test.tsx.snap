// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test matches Snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "header header--internal",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "header__top",
          "role": "banner",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "container",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "header__grid-top grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap ",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "grid__col col-auto",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "class": "header__logo-link",
                          "href": "/",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {},
                            "children": Array [
                              Object {
                                "attribs": Object {
                                  "alt": "Office for National Statistics logo",
                                  "class": "header__logo",
                                  "src": "https://ons-design-system.netlify.app/img/ons-logo-neg-en.svg",
                                },
                                "children": Array [],
                                "name": "img",
                                "namespace": "http://www.w3.org/1999/xhtml",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "tag",
                                "x-attribsNamespace": Object {
                                  "alt": undefined,
                                  "class": undefined,
                                  "src": undefined,
                                },
                                "x-attribsPrefix": Object {
                                  "alt": undefined,
                                  "class": undefined,
                                  "src": undefined,
                                },
                              },
                            ],
                            "name": "picture",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {},
                            "x-attribsPrefix": Object {},
                          },
                        ],
                        "name": "a",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                          "href": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                          "href": undefined,
                        },
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "header__main",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "container",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "grid__col col-auto u-flex-shrink",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "class": "header__title",
                          },
                          "children": Array [
                            Object {
                              "data": "App Title",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "div",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "class": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "class": undefined,
                          },
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "role": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "role": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "header__main",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "container",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "grid__col col-auto u-flex-shrink",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "class": "header__title",
                        },
                        "children": Array [
                          Object {
                            "data": "App Title",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "div",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                        },
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "header__top",
            "role": "banner",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "container",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "header__grid-top grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap ",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "grid__col col-auto",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "class": "header__logo-link",
                            "href": "/",
                          },
                          "children": Array [
                            Object {
                              "attribs": Object {},
                              "children": Array [
                                Object {
                                  "attribs": Object {
                                    "alt": "Office for National Statistics logo",
                                    "class": "header__logo",
                                    "src": "https://ons-design-system.netlify.app/img/ons-logo-neg-en.svg",
                                  },
                                  "children": Array [],
                                  "name": "img",
                                  "namespace": "http://www.w3.org/1999/xhtml",
                                  "next": null,
                                  "parent": [Circular],
                                  "prev": null,
                                  "type": "tag",
                                  "x-attribsNamespace": Object {
                                    "alt": undefined,
                                    "class": undefined,
                                    "src": undefined,
                                  },
                                  "x-attribsPrefix": Object {
                                    "alt": undefined,
                                    "class": undefined,
                                    "src": undefined,
                                  },
                                },
                              ],
                              "name": "picture",
                              "namespace": "http://www.w3.org/1999/xhtml",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "tag",
                              "x-attribsNamespace": Object {},
                              "x-attribsPrefix": Object {},
                            },
                          ],
                          "name": "a",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "class": undefined,
                            "href": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "class": undefined,
                            "href": undefined,
                          },
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "role": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "role": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "header",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
